core/camel-core/src/test/java/org/apache/camel/impl/JsonApiDataFormatTest.java

```diff
@@ -0,0 +1,149 @@
+
+        Exchange exchange = new DefaultExchange(context);
+        ByteArrayOutputStream baos = new ByteArrayOutputStream();
+        jsonApiDataFormat.marshal(exchange, book, baos);
+
+        String jsonApiOutput = baos.toString();
+        assertNotNull(jsonApiOutput);
+        assertEquals(this.generateTestDataAsString(), jsonApiOutput);
+    }
+
+    @Test(expected = DocumentSerializationException.class)
+    public void test_jsonApi_marshal_no_annotation_on_type() throws Exception {
+        Class<?>[] formats = { MyBook.class, MyAuthor.class };
+        jsonApiDataFormat = new JsonApiDataFormat();
+        jsonApiDataFormat.setDataFormatTypes(formats);
+
+        Exchange exchange = new DefaultExchange(context);
+        ByteArrayOutputStream baos = new ByteArrayOutputStream();
+        jsonApiDataFormat.marshal(exchange, new FooBar(), baos);
+    }
+
+    @Test(expected = DocumentSerializationException.class)
+    public void test_jsonApi_marshal_wrong_type() throws Exception {
+        Class<?>[] formats = { MyBook.class, MyAuthor.class };
+        jsonApiDataFormat = new JsonApiDataFormat();
+        jsonApiDataFormat.setDataFormatTypes(formats);
+
+        Exchange exchange = new DefaultExchange(context);
+        ByteArrayOutputStream baos = new ByteArrayOutputStream();
+        jsonApiDataFormat.marshal(exchange, new MyFooBar("bar"), baos);
+    }
+
+    @Test
+    public void test_jsonApi_unmarshal() throws Exception {
+        Class<?>[] formats = { MyBook.class, MyAuthor.class };
+        jsonApiDataFormat = new JsonApiDataFormat();
+        jsonApiDataFormat.setDataFormatTypes(formats);
+        jsonApiDataFormat.setMainFormatType(MyBook.class);
+
+        String jsonApiInput = this.generateTestDataAsString();
+
+        Exchange exchange = new DefaultExchange(context);
+        Object outputObj = jsonApiDataFormat.unmarshal(exchange, new ByteArrayInputStream(jsonApiInput.getBytes()));
+
+        assertNotNull(outputObj);
+        MyBook book = (MyBook)outputObj;
+        assertEquals("Camel in Action", book.getTitle());
+        assertEquals("1", book.getAuthor().getAuthorId());
+    }
+
+    @Test(expected = UnregisteredTypeException.class)
+    public void test_jsonApi_unmarshal_wrong_type() throws Exception {
+        Class<?>[] formats = { MyBook.class, MyAuthor.class };
+        jsonApiDataFormat = new JsonApiDataFormat();
+        jsonApiDataFormat.setDataFormatTypes(formats);
+        jsonApiDataFormat.setMainFormatType(MyBook.class);
+
+        String jsonApiInput = "{\"data\":{\"type\":\"animal\",\"id\":\"camel\",\"attributes\":{\"humps\":\"2\"}}}";
+
+        Exchange exchange = new DefaultExchange(context);
+        jsonApiDataFormat.unmarshal(exchange, new ByteArrayInputStream(jsonApiInput.getBytes()));
+    }
+
+    @Ignore
+    public void test_jsonApi_with_route() {
+        final String title = "Hello Thai Elephant \u0E08";
+
+        // context.addRoutes(new RouteBuilder() {
```

Comment:
Can be removed

Reply:
of course. Was hoping to uncomment this once had some pointers on how make the DataFormat accessible in the route.